src/app/catalog/services/catalog/catalog-filter.service.ts

```diff
@@ -1,15 +1,20 @@
 import { Injectable } from '@angular/core';
-import { filter, Observable, Subject } from 'rxjs';
+import { of, Observable, Subject, switchMap } from 'rxjs';
 import { FilterModalComponent } from '@app/catalog/components/filter-modal/filter-modal.component';
 import { MatDialog } from '@angular/material/dialog';
+import { EquipmentFilter, EquipmentFilterRequest } from '@app/catalog/models';
+import { Store } from '@ngxs/store';
+import { CatalogState, SetEquipmentFilter, SetSearchInput, SetSelectedCategoryId } from '@app/catalog/store';
+import { ApplicationDataState } from '@shared/store/application-data';
 
 @Injectable({
   providedIn: 'root',
 })
 export class CatalogFilterService {
   private filtersButtonDisplayed = new Subject<boolean>();
+  private filteringApplied = new Subject<boolean>();
```

Comment:
since we have store and since we have selectedCategoryId select  we do not need this. we can subscribe select whenever we want and have the same result.